text/0005-application-formalization.md

```diff
@@ -0,0 +1,191 @@
+      "name": "app-organizations",
+      "version": "^0.0.1"
+    }
+  ],
+  "modules": [
+    {
+      "id": "mod-gobi-2.6.0",
+      "name": "mod-gobi",
+      "version": "2.6.0",
+	  "url": "https://folio-registry.dev.folio.org/_/proxy/modules/mod-gobi-2.6.0"
+    }
+  ],
+  "uiModules": [
+    {
+      "id": "folio_gobi-settings-2.0.0",
+      "name": "folio_gobi-settings",
+      "version": "2.0.0",
+	  "url": "https://folio-registry.dev.folio.org/_/proxy/modules/folio_gobi-settings-2.0.0"
+    }
+  ]
+}
+```
+**N.B.** Full module descriptors and metadata have been omitted for readability reasons.
+
+## Benefits
+* Allows for expanding the bounded context
+  * By expanding the bounded context from the module to the Application, multiple modules can directly access the storage of the microservice.
+  * Benefites of this include:  
+    * Improved performance and simplified logic.  Modules can now perform the same operations with fewer API calls.
+    * Fewer modules since this eliminates the need for separate business logic and storage modules.
+* This is the first step toward breaking the monolithic release into application-specific releases
+* Results in fewer things to work with at the system operator level
+* Brings us closer to realizing the idea of having an application store/marketplace
+
+## Risks and Drawbacks
+* Folio has too many module dependencies which must be reduced
+* Transition to Applications will need to be iterative
+  * First pass is to create larger applications spanning multiple areas in order to satisfy dependencies
+  * Subsequent passes will further break up these large applications into smaller, applications focused on a single area of functionality
+* Since Applications are a vertical slice of functionality, better alignment/coordination between back-end and front-end development teams is required.
+  * Reorganizing development teams around applications would help
+* By latching onto the Application term, there may be confusion about terminology since that term currently means different things to different people.
+  * It may help to proposed formal names for other things currently referred to as Applications (e.g. the icons that appear in the Folio toolbar)
+
+## Rationale and Alternatives
+Alternative approaches include:
+* **Status quo** - Don't do anything and deal with monolithic releases, an ever-growing number of modules, and all the problems and challenges associated with those things.
+* **Independent module release cycles** - Impractical due to how tightly coupled modules are to one another.   It also doesn't address the issue of system operators needing to manage an ever growing number of modules.
+
+## Unresolved Questions
+The following unresolved questions are not addressed here, but will be the addressed in other RFCs:
+* How will Application management work?
+* How will Application Descriptors be developed and maintained?
+* How the Stripes UI will be updated as applications are enabled/disabled?
```

Comment:
By "enabled/disabled" does this mean the user-activated application **upgrade** process?  If not then it should be listed separately as there are many unresolved questions on that.

Reply:
Applications are enabled/disabled for tenants just like modules are today.  "Upgrading" an application is essentially a convenient alternative to explicitly disabling application v1, and enabling application v2.  All of the above (explicit enable, disable, upgrade) are applicable to the text in question.

I've added `/upgraded` to `enabled/disabled`.